src/Parallel/ParallelComponentHelpers.hpp

```diff
@@ -293,6 +294,51 @@ using get_option_tags = tmpl::remove_duplicates<tmpl::flatten<
                                    Metavariables>::template f,
                                tmpl::_1>>>>;
 
+namespace detail {
+template <typename Tag, typename = std::void_t<>>
+struct is_tag_overlayable : std::false_type {};
+
+template <typename Tag>
+struct is_tag_overlayable<Tag, std::void_t<decltype(Tag::is_overlayable)>>
+    : std::bool_constant<Tag::is_overlayable> {
+  static_assert(tmpl::size<typename Tag::option_tags>::value == 1,
```

Comment:
Include `not is_overlayable or` in this.  Although it looks like you have a similar assertion where this restriction is actually assumed, so maybe just remove this one and rely on the check there.